src/nnet3/nnet-training.cc

```diff
@@ -235,6 +235,45 @@ void ComputeObjectiveFunction(const GeneralMatrix &supervision,
         computer->AcceptOutputDeriv(output_name, &diff);
       break;
     }
+    case kLognormal: {
```

Comment:
I think the same. Other than that, this objective function is very specific and might not be used in any other task. I just implemented it like this so that I could try it quickly. If it turns out to be useful (and you decide to keep the duration modeling feature) we can find a proper place for it.